ttnn/cpp/ttnn/operations/moreh/moreh_dot_op_backward/moreh_dot_backward.cpp

```diff
@@ -7,14 +7,36 @@
 #include "ttnn/operations/moreh/moreh_dot_op_backward/device/moreh_dot_backward_device_operation.hpp"
 
 namespace ttnn::operations::moreh::moreh_dot_backward {
-std::vector<Tensor> MorehDotBackward::invoke(
+std::vector<std::optional<Tensor>> MorehDotBackward::invoke(
     const Tensor &output_grad,
     const Tensor &input,
     const Tensor &other,
-    std::optional<const Tensor> input_grad,
-    std::optional<const Tensor> other_grad,
+    const std::optional<const Tensor> &input_grad,
+    const std::optional<const Tensor> &other_grad,
     const std::optional<MemoryConfig> &mem_config) {
-    return ttnn::prim::moreh_dot_backward(
-        output_grad, input, other, input_grad, other_grad, mem_config);
+    return ttnn::prim::moreh_dot_backward(output_grad, input, other, input_grad, other_grad, mem_config);
 }
-}  // namespace ttnn::operations::moreh::moreh_matmul
+
+std::vector<Tensor> create_async_output_tensors(
+    const std::vector<Tensor> &input_tensors, const std::vector<std::optional<Tensor>> &optional_inputs) {
+    auto output_grad = input_tensors.at(0);
+    auto input = input_tensors.at(1);
+    auto other = input_tensors.at(2);
+
+    return {
+        Tensor(operation::get_workers_for_op_output({output_grad, input, other})),
+        Tensor(operation::get_workers_for_op_output({output_grad, input, other})),
+    };
+}
+
+std::vector<bool> create_async_tensor_flag(
+    const Tensor &output_grad,
+    const Tensor &input,
+    const Tensor &other,
+    const std::optional<const Tensor> &input_grad,
+    const std::optional<const Tensor> &other_grad,
+    const std::optional<MemoryConfig> &mem_config) {
+    return {true, true};
```

Comment:
if its always all true you should return return `vector<Tensor>` instead of `vector<optional<Tensor>>`

Reply:
Probably something I didn't notice while editing this ops (it was someone else's). Anyway, fixed.